app/uk/gov/hmrc/exports/controllers/NotificationController.scala

```diff
@@ -42,17 +41,27 @@ class NotificationController @Inject()(
   headerValidator: HeaderValidator,
   metrics: ExportsMetrics,
   notificationsService: NotificationService,
+  submissionService: SubmissionService,
   bodyParsers: PlayBodyParsers,
   cc: ControllerComponents
-)(implicit executionContext: ExecutionContext) extends Authenticator(authConnector, cc) {
+)(implicit executionContext: ExecutionContext) extends Authenticator(authConnector, cc) with JSONResponses {
 
   private val logger = Logger(this.getClass)
 
+  def findByID(id: String): Action[AnyContent] = authorisedAction(bodyParsers.default) { implicit request =>
+    submissionService.getSubmission(id) flatMap {
+      case Some(submission) if submission.mrn.isDefined =>
+        notificationsService.getNotificationsForSubmission(submission.mrn.get)
+          .map(notifications => Ok(notifications))
+      case _ => Future.successful(NotFound)
+    }
+  }
+
   def getSubmissionNotifications(mrn: String): Action[AnyContent] =
     authorisedAction(bodyParsers.default) { implicit request =>
       notificationsService
         .getNotificationsForSubmission(mrn)
-        .map(notifications => Ok(Json.toJson(notifications)))
```

Comment:
And I prefer previous explicit `Json.toJson` I'm not sure that you're solving some problem or it's just improvement to write less code

Reply:
Personally if I don't have to write `Json.toJson` each time its better

Reply:
Yeah, but for new Controller you need to remember about this Trait, I'm not sure about it, let's wait for other people :) I don't want to block PR :)

Reply:
New controllers will extend the RESTController trait which pulls it in for free. I didnt want to change this one as it's likely to be removed once we stop using conversation-id everywhere

Reply:
I agree with Patryk, its not obvious you need to include this trait. Also the Json.toJson whilst more verbose shows there is a conversion actually happening. We are invoking the Json writer here. Removing it hides that and it all just happens magically which reads nicer but could argue is more confusing.

Reply:
For me less boiler plate is better, Play is the only framework I've seen that makes you manually convert all payloads to & from strings, others e.g. SpringBoot, DropWizard presume you want JSON and do it for you.
 
Regarding creating new controllers, it's easy to miss a lot of things so we'd generally copy and paste an existing one anyway. Any new Controllers would extend `RESTController` which gives everything for free

Reply:
If the DeclarationController (which extends RESTController) is the accepted "new" way of creating json responses then I see no reason why the Notification and Submission controllers shouldn't (temporarily?) do the same thing.